samza-api/src/main/java/org/apache/samza/operators/windows/WindowKey.java

```diff
@@ -52,4 +52,29 @@ public String toString() {
     }
     return String.format("%s%s", wndKey, paneId);
   }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
Good point. I modified the docs to have more details (on what `paneId` means for session and tumbling windows).